hangman.py

```diff
@@ -0,0 +1,322 @@
+    """
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    """
+    tries = INITIAL_TRIES
+    warn = INITIAL_WARN
+    # List "letters_guessed" is used as a container for letters which have already been guessed
+    letters_guessed = set()
+    # Variable "word" shows the length of the secret word and which letters have already been guessed
+    word = get_guessed_word(secret_word, letters_guessed)
+    start_message(secret_word, hints)
+
+    while tries != 0 and is_word_guessed(secret_word, word):
+        letter = game_round(letters_guessed, tries)
+        if len(letter) == 1:
+            # If the entered symbol is a "*", the program uses a hint and shows all possible matches
+            if hints and letter == "*":
+                show_possible_matches(word)
+            else:
+                tries, warn, letters_guessed, word = check_element(tries, warn, letters_guessed, word, letter)
+
+        # If user enters more or less than one symbol, the program issues a warning
+        else:
+            warn, tries = spend_warn(warn, tries)
+            print(f"Oops! Don't enter more or less than one symbol. You have {warn if warn >= 0 else 'no'} "
+                  f"warnings left{' so you lose one guess' if warn < 0 else ''}: {word}")
+
+    end_message(secret_word, tries)
+    return
+
+
+def match_with_gaps(my_word, other_word):
+    """
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    """
+    # This loop compares my_word, which consists of "_ " and letters which have already been guessed, with words
+    # from file "words.txt". If these two words have the same length and letters from my_word have the same
+    # position in the compared word, loop returns True. Otherwise, it returns False.
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if my_word[i] != "_" and (
```

Comment:
This '_' symbol, as well as '*', has some meaning in that game, so maybe it makes sense to create a constant for it as well?